cypress/support/commands/application.js

```diff
@@ -1,13 +1,45 @@
-Cypress.Commands.add('testResponseCodeAndType', (path, responseCode, type) => {
-  cy.request({ url: path, failOnStatusCode: false }).then(
-    ({ status, headers }) => {
-      expect(status).to.eq(responseCode);
-      expect(headers['content-type']).to.include(type);
-      // Always ensure we're not seeing the Mozart fallback
-      expect(
-        headers,
-        `Mozart fallback response detected for ${path}`,
-      ).not.to.have.property('x-mfa');
-    },
-  );
-});
+// Custom command to test that the specified path returns the expected status code and
+// content type, as well as that's the response was not a Mozart fallback. Automatically
+// retries twice after a delay if it gets an unexpected response
+// NB:
+// - Default timeout for cy.request is 50s
+// - Certain types of network error are retried automatically (retryOnNetworkFailure)
+Cypress.Commands.add(
+  'testResponseCodeAndType',
+  (path, responseCode, type, retriesLeft = 2) => {
```

Comment:
Not sure if it would be beneficial, but instead of a default value for the retry parameter could we make it a required argument for the function call? Just in case, in the future use of this command you can explicitly set the amount times you would like it to retry.

Reply:
My thinking here was that most people will be happy to accept the default of three tries and so it reduces the amount of typing  whilst still allowing people to request more or less tries if they want to by passing a value for this argument. Does that make sense? If not let me know and we can jump on a call to chat.

Reply:
Nope, it's fine that makes sense. 👍